pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -145,6 +145,16 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 			return errors.Wrap(err, "failed to create crash collector kubernetes secret")
 		}
 	}
+	// Create exporter secret
+	if !cluster.Spec.Monitoring.MetricsDisabled {
+		if cluster.ClusterInfo.CephCred.Username == client.AdminUsername {
```

Comment:
Because of some legacy code the admin name is added to both   the cases, so validate using 

`clusterInfo.CephCred.Username == cephclient.AdminUsername && clusterInfo.CephCred.Secret != adminSecretNameKey`

Please validate the non-admin scenario too, as the legacy secrets are two complicated here

Reply:
Done with new commit

Reply:
No secret is created in both the scenario, but with min privileges

Reply:
As there is a legacy code, and most user run with non privilege env, so it would be better to quickly update the caps and test that way